### DISCUSSION TITLE
大佬 想问一下Dxcam模式好像黑屏

### DISCUSSION THREAD

**Discussion by kujoykarendesu** [Passer1072/RookieAI_yolov8#28] [Q&A]
rt,我是笔记本3060 显卡驱动用的还是GeForce的 不知道有没有影响？ 还是说要去下一个cuda驱动，我现在dxcam打开的话 有帧数 但是就是黑屏 不截图 想请教一下这个是啥问题呢 有什么解决办法吗
昨天我让gpt优化了一下那个 随机特征码的 我不知道这样子会不会更有效果我也顺便发发 但是我python学习不久 很多东西都没接触过 我只是CV出来我不知道具体有没有用
import string
import random
import hashlib
import platform
import socket
import ctypes
======================
def generate_fingerprint():  # 随机特征码
    # 收集系统信息
    system_info = {
        'platform': platform.platform(),
        'machine': platform.machine(),
        'processor': platform.processor(),
        'hostname': socket.gethostname(),
        'ip_address': socket.gethostbyname(socket.gethostname()),
        'username': os.getenv('USER') or os.getenv('USERNAME'),
        'home': os.path.expanduser('~'),
        'random_uuid': str(uuid.uuid4()),
        'timestamp': str(datetime.datetime.now()),
    }

    # 将所有信息转换为字符串并连接
    fingerprint_str = ''.join(f"{k}:{v}" for k, v in system_info.items())

    # 使用SHA256生成哈希
    return hashlib.sha256(fingerprint_str.encode()).hexdigest()

def random_string(length):  # 随机软件名称
    # 使用更安全的随机源
    return ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits) for _ in range(length))
顺便想问问大佬有买kmboxnet吗 我现在也打算入一个kmboxnet 我想看看大佬后续的开发进展是什么样 我想继续跟进大佬的项目来学习  所以想问问

**Comment by Passer1072** [comment]
关于驱动：正常使用游戏驱动就可以了，如果C盘空间充足可以考虑下载[CUDA工具包](https://developer.nvidia.com/cuda-toolkit)在某些地方会有些用。
关于DXcam黑屏问题：大概率是环境问题，之前我也出现了但是某次重新整理exe文件的运行环境后就好了，你可下载[成品](https://1drv.ms/u/s!At2Twb-tSku-lsoHEql6iAi-X5eZNQ?e=kTUH8X)直接运行里面的exe文件尝试能不能正常使用DXcam模式，如果可以的话对照下_internal文件夹内的各种库的版本是否一致，我是Python3.9。这个只能自己排查了。完整[requirements.txt](https://github.com/Passer1072/RookieAI_yolov8/blob/main/Beta/requirements.txt)
关于特征码：这个可以根据自己见解随意更改不是很重要，最好不要收集本机的任何信息。
关于kmbox：计划采购kmbox net版本，之后也只会对net版本做适配，B+和普通版暂不考虑。你可以先买着自己试着做下适配，我什么时候能做出来我也不知道 XD

**Comment by kujoykarendesu** [comment]
好的谢谢大佬,我昨天晚上试了一下好像我用成品DXCAM也黑屏 然后我昨天晚上又重新安装了一次python成3.9 pip install了 你readme里面的那几行环境 不过还是会黑屏 不过有MSS截图也用不上DXCAM就是了
KMBOX好像是有固件和开什么曲线这个说法的，好像硬件到手还是需要去官网刷新的固件？ 不过想问一下大佬GITHUB或者是大佬你手上有支持KMBOX的AI源码吗 不然感觉我都不知道怎么实现KMBOX调用 我现在还没买盒子 也不知道KMBOX有没有那种教学案例...

**Reply by Passer1072** [reply]
> 好的谢谢大佬,我昨天晚上试了一下好像我用成品DXCAM也黑屏 然后我昨天晚上又重新安装了一次python成3.9 pip install了 你readme里面的那几行环境 不过还是会黑屏 不过有MSS截图也用不上DXCAM就是了 KMBOX好像是有固件和开什么曲线这个说法的，好像硬件到手还是需要去官网刷新的固件？ 不过想问一下大佬GITHUB或者是大佬你手上有支持KMBOX的AI源码吗 不然感觉我都不知道怎么实现KMBOX调用 我现在还没买盒子 也不知道KMBOX有没有那种教学案例...

我手上没有，但是KMBOX[官网](https://www.kmbox.top/)好像有Python调试教程和demo可以自己先研究下。

**Reply by kujoykarendesu** [reply]
好的 谢谢
